client/client.go

```diff
@@ -206,6 +207,16 @@ func (c *LogClient) queueLeaf(ctx context.Context, leaf *trillian.LogLeaf) error
 		LogId: c.LogID,
 		Leaf:  leaf,
 	}
-	_, err := c.client.QueueLeaf(ctx, &req)
-	return err
+	rsp, err := c.client.QueueLeaf(ctx, &req)
+	if err != nil {
+		return err
+	}
+	if rsp == nil {
+		return errors.New("no QueueLeafResponse generated")
+	}
+	if rsp.QueuedLeaf.Status.Code == int32(code.Code_ALREADY_EXISTS) {
```

Comment:
That would still need a cast, wouldn't it? 
- `grpc.ALREADY_EXISTS` is a `grpc.Code` which is a `uint32`
- `rpc.ALREADY_EXISTS` is an `rpc.Code` which is an `int32`
- `rpc.Status.Code` is of type `int32`